pkg/validating-webhooks/hiveextension/v1beta1/agentclusterinstall_admission_hook.go

```diff
@@ -0,0 +1,292 @@
+	contextLogger := log.WithFields(log.Fields{
+		"operation": admissionSpec.Operation,
+		"group":     admissionSpec.Resource.Group,
+		"version":   admissionSpec.Resource.Version,
+		"resource":  admissionSpec.Resource.Resource,
+		"method":    "Validate",
+	})
+
+	if !a.shouldValidate(admissionSpec) {
+		contextLogger.Info("Skipping validation for request")
+		// The request object isn't something that this validator should validate.
+		// Therefore, we say that it's allowed.
+		return &admissionv1.AdmissionResponse{
+			Allowed: true,
+		}
+	}
+
+	contextLogger.Info("Validating request")
+
+	if admissionSpec.Operation == admissionv1.Create {
+		return a.validateCreate(admissionSpec)
+	}
+
+	if admissionSpec.Operation == admissionv1.Update {
+		return a.validateUpdate(admissionSpec)
+	}
+
+	// We're only validating creates and updates at this time, so all other operations are explicitly allowed.
+	contextLogger.Info("Successful validation")
+	return &admissionv1.AdmissionResponse{
+		Allowed: true,
+	}
+}
+
+// shouldValidate explicitly checks if the request should validated. For example, this webhook may have accidentally been registered to check
+// the validity of some other type of object with a different GVR.
+func (a *AgentClusterInstallValidatingAdmissionHook) shouldValidate(admissionSpec *admissionv1.AdmissionRequest) bool {
+	contextLogger := log.WithFields(log.Fields{
+		"operation": admissionSpec.Operation,
+		"group":     admissionSpec.Resource.Group,
+		"version":   admissionSpec.Resource.Version,
+		"resource":  admissionSpec.Resource.Resource,
+		"method":    "shouldValidate",
+	})
+
+	if admissionSpec.Resource.Group != agentClusterInstallGroup {
+		contextLogger.Debug("Returning False, not our group")
+		return false
+	}
+
+	if admissionSpec.Resource.Version != agentClusterInstallVersion {
+		contextLogger.Debug("Returning False, it's our group, but not the right version")
+		return false
+	}
+
+	if admissionSpec.Resource.Resource != agentClusterInstallResource {
+		contextLogger.Debug("Returning False, it's our group and version, but not the right resource")
+		return false
+	}
+
+	// If we get here, then we're supposed to validate the object.
+	contextLogger.Debug("Returning True, passed all prerequisites.")
+	return true
+}
+
+// validateCreate specifically validates create operations for AgentClusterInstall objects.
+func (a *AgentClusterInstallValidatingAdmissionHook) validateCreate(admissionSpec *admissionv1.AdmissionRequest) *admissionv1.AdmissionResponse {
+	contextLogger := log.WithFields(log.Fields{
+		"operation": admissionSpec.Operation,
+		"group":     admissionSpec.Resource.Group,
+		"version":   admissionSpec.Resource.Version,
+		"resource":  admissionSpec.Resource.Resource,
+		"method":    "validateCreate",
+	})
+
+	aci := &hiveext.AgentClusterInstall{}
+	if err := a.decoder.DecodeRaw(admissionSpec.Object, aci); err != nil {
```

Comment:
Are we just testing that we can unmarshal the object? Do we need to validate creation at all then?

Reply:
Agree, no validation on Create for now, so I will remove this.

Reply:
Done